File: 프로그래머스/lv1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        String answer = "";
        String[] bowl = s.split("");
        boolean trust = true;
        for(int i = 0; i < bowl.length; i++){
            if(!" ".equals(bowl[i])){
                if(trust){
                    bowl[i] = bowl[i].toUpperCase();
                    trust = false;
                }else{
                    bowl[i] = bowl[i].toLowerCase();
                    trust = true;
                }
            }else{
                trust = true;
            }
        }
        return answer = String.join("", bowl);
    }
}

File: 프로그래머스/lv1/12930. 이상한 문자 만들기/이상한 문자 만들기.js
function solution(s) {
    var answer = s.split("");
    let cnt = 0;
    for(let i = 0; i < answer.length; i++){
        if(answer[i] != " "){
            console.log("answer[i] = " + answer[i] + " cnt = " + cnt);
            if(cnt == 0){
                answer[i] = answer[i].toUpperCase();
                cnt++;
            }else{
                answer[i] = answer[i].toLowerCase();
                cnt--;
            }    
        }else{
            cnt = 0;
        }
    }
    return answer.join("");
}